powerhome/playbook
# [NUXE-331] Kit search

#### What does this PR do?

1. Creates a search bar for kits
2. Reimagines top navbar to help accomodate kit search (new dark mode toggle, smaller nav kit).
3. Adds new horizontal subtle variant to nav kit.
4. Improves doc site React component file structure and js pipeline.
5. Resolves conflicting duplicate WebpackerReact registrations.
6. Cleans up js packs includes.

**Bonus**

7. Fixes bug where dark mode cookie carries over to Visual Guidelines page.  Moves cookie deleting method into `pb_doc_helper` so it can be used in multiple controllers.

#### Screens

<img width="988" alt="top_nav_search" src="https://user-images.githubusercontent.com/57693937/101790515-89890900-3ad0-11eb-8e40-1e0839bb55f7.png">
<img width="1003" alt="light_search" src="https://user-images.githubusercontent.com/57693937/101790514-89890900-3ad0-11eb-9539-954e73c8f884.png">
<img width="1001" alt="dark_search" src="https://user-images.githubusercontent.com/57693937/101790512-89890900-3ad0-11eb-8cfc-4320ec6a5bca.png">

#### Breaking Changes

Some changes to controller logic/helpers but shouldn't break any existing functionality on site.

#### Runway Ticket URL

[Link to Runway Story](https://nitro.powerhrg.com/runway/backlog_items/NUXE-331)

#### How to test this

Test out new features in Milano deploy.  Make sure mobile sizing works as expected.

#### Checklist:


*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*
